job/server/src/main/java/alluxio/job/plan/migrate/MigrateDefinition.java

```diff
@@ -181,8 +181,7 @@ private void checkMigrateValid(MigrateConfig config, FileSystem fs) throws Excep
     for (WorkerInfo workerInfo : jobWorkerInfoList) {
       hostnameToWorker.put(workerInfo.getAddress().getHost(), workerInfo);
     }
-    List<BlockWorkerInfo> alluxioWorkerInfoList =
-        AlluxioBlockStore.create(context.getFsContext()).getAllWorkers();
+    List<BlockWorkerInfo> alluxioWorkerInfoList = context.getFsContext().getAllWorkers();
```

Comment:
Change all the getAllWorkers to getEligibleWorkers?

i tested locally that if all job servers used `getEligibleWorkers()` instead of `getAllWorkers()`, a large amount of `getWorkerInfoList` call will be reduced on the job server side. 

This is because instead of creating a `AlluxioBlockStore` each time when needed, Those context used in job server definitions and job utils internally contain a shared `JobServerContext` which contains a `FileSystemContext`.
This FileSystemContext is shared so that the cached worker info list is shared.

Reply:
agree